main.go

```diff
@@ -20,20 +21,78 @@ import (
 	"gopkg.in/matryer/try.v1"
 )
 
+const (
+	connectionTimeoutEnv string = "RYKU_CONNECTION_TIMEOUT"
+)
+
 var (
-	port                  = flag.Int("p", 8080, "Port to bind at")
-	initialConnectTimeout = 1 * time.Minute
-	reconnectionTimeout   = 10 * time.Second
+	port                int
+	connectionTimeout   time.Duration
+	reconnectionTimeout time.Duration
 )
 
+type config struct {
+	Port                int
+	ConnectionTimeout   time.Duration
+	ReconnectionTimeout time.Duration
+}
+
+// newConfig parses command line flags and returns a parsed config. config.timeout
+// can be set by environment variable, RKYU_CONNECTION_TIMEOUT. If both flag and
+// environment variable are defined the flag is prioritized. If an error occurs
+// while parsing RKYU_CONNECTION_TIMEOUT the error is returned.
+func newConfig(args []string) (*config, error) {
+	cfg := config{ReconnectionTimeout: 10 * time.Second}
+
+	fs := flag.NewFlagSet("ryku", flag.PanicOnError)
```

Comment:
should the FlagSet name be `ryku` or `moby-ryku`?

Reply:
We use `ryuk.container.timeout` as the property in Java.

Reply:
This references the name of the program which called by the command line. In `main` it shows the following:
```
$ ./bin/moby-ryuk -h
Usage of ./bin/moby-ryuk:
  -p int
        Port to bind at (default 8080)
```

With the flagset:
```
$ ./bin/moby-ryuk -h
Usage of ryuk:
  -p int
        Port to bind at (default 8080)
  -timeout duration
        Connection timeout (default 1m0s)
```